runtime/service.go

```diff
@@ -265,11 +304,18 @@ func (s *service) Resume(ctx context.Context, req *taskAPI.ResumeRequest) (*ptyp
 // Kill a process with the provided signal
 func (s *service) Kill(ctx context.Context, req *taskAPI.KillRequest) (*ptypes.Empty, error) {
 	log.G(ctx).WithFields(logrus.Fields{"id": req.ID, "exec_id": req.ExecID}).Debug("kill")
+	// right now we want to kill vm always when kill is called
+	// may not be true in multi-container vm
+	defer func() {
+		log.G(ctx).Debug("Stopping VM during kill")
+		if err := s.stopVM(); err != nil {
```

Comment:
`stopVM()` simply terminates the firecracker process. We need to take care to shut down more cleanly, or else we risk corrupting the VMs filesystem and/or losing buffered writes.

We probably want to run `/sbin/reboot` inside the VM during agent's handling of `Shutdown` and `Kill`, then wait a short period of time before terminating the VM more forcefully (if it still exists).

Reply:
I think this makes sense.  I think there are some other shutdown related issues that should be dealt with together. I created #25 to track that. Do you feel strongly about it being done in this PR?

Reply:
OK.